packages/core/src/pipeline/config/stages/bakeManifest/BakeManifestDetailsTab.tsx

```diff
@@ -3,31 +3,43 @@ import React from 'react';
 import type { IExecutionDetailsSectionProps } from '../common';
 import { ExecutionDetailsSection } from '../common';
 import { StageFailureMessage } from '../../../details';
-import type { IArtifact } from '../../../../domain';
+import { ARTIFACT_TYPE_EMBEDDED } from '../../../../domain';
 import { ManifestYaml } from '../../../../manifest';
 import { Overridable } from '../../../../overrideRegistry';
 import { decodeUnicodeBase64 } from '../../../../utils';
+import { getBakedArtifacts } from './utils/getBakedArtifacts';
+import { getContentReference } from './utils/getContentReference';
 
 @Overridable('bakeManifest.bakeManifestDetailsTab')
 export class BakeManifestDetailsTab extends React.Component<IExecutionDetailsSectionProps> {
   public static title = 'bakedManifest';
 
   public render() {
     const { current, name, stage } = this.props;
-    const bakedArtifacts: IArtifact[] = (stage.context.artifacts || []).filter(
-      (a: IArtifact) => a.type === 'embedded/base64',
-    );
+    const bakedArtifacts = getBakedArtifacts(stage.context);
+
     return (
       <ExecutionDetailsSection name={name} current={current}>
         <StageFailureMessage stage={stage} message={stage.failureMessage} />
-        {bakedArtifacts.map((artifact, i) => (
-          <ManifestYaml
-            key={i}
-            linkName={bakedArtifacts.length > 1 ? `Baked Manifest ${i} YAML` : 'Baked Manifest YAML'}
-            manifestText={decodeUnicodeBase64(artifact.reference)}
-            modalTitle="Baked Manifest"
-          />
-        ))}
+        {bakedArtifacts.map((artifact, i) => {
+          const linkName = bakedArtifacts.length > 1 ? `Baked Manifest ${i} YAML` : 'Baked Manifest YAML';
```

Comment:
I haven't had a chance to review the whole set of PRs - can a bake stage now bake multiple times?

Reply:
This is old code from 4 years ago. We just moved it into a variable.